File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int low = 0;
        int high = nums.length - 1;

        while (low <= high) {
            int mid = (low + high) / 2;

            if (nums[mid] == target) {
                return mid;
            } else if (target > nums[mid]){
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }
        return -1;
    }
}

// have low, high, and mid
// low is first in array
// high his last in array
// mid is int array length/2 

// do while mid != target

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        j = 0
        k = len(nums) - 1

        while j < k:
            mid = (k + j) // 2

            if nums[mid] > nums[k]:
                j = mid + 1
            else: 
                k = mid
        return nums[j]